editor/create_dialog.cpp

```diff
@@ -73,7 +73,8 @@ void CreateDialog::popup_create(bool p_dont_clear, bool p_replace_mode, const St
 	if (saved_size != Rect2()) {
 		popup(saved_size);
 	} else {
-		popup_centered_clamped(Size2(900, 700) * EDSCALE, 0.8);
+		Size2i popup_size = get_screen_transform().basis_xform(Vector2i(900 * EDSCALE, 700 * EDSCALE));
+		popup_centered_clamped(popup_size, 0.8);
```

Comment:
I'm not entirely familiar with `EDSCALE` and the per-window-scaling.
I would appreciate, if you could tell me, if my interpretation of this code section is correct:

The intention of the code is to create a Window with a viewport size of 900x700 pixel. This size is adjusted by EDSCALE. So EDSCALE=2 means that the windows viewport should have a size of 1800x1400.

In order to allow per-window-scaling, it becomes necessary, that the each window can additionally be adjusted by a factor.
When the per-window-scaling has for example a value of `1.5`, then the effect is, that a window with a 1800x1400 viewport takes up 2700x2100 pixel on the screen (or its embedder if embedded).

Reply:
Yes, your calculations are correct.

EDSCALE is a user adjustable scale that is applied on top of window scale (and it's global, not pre-window/display). It's currently used to adjust editor size to DPI in some cases, but with window scaling should be `1` in the majority of cases.

Reply:
Thank you for the quick reply! Then the above code section should be changed to

```suggestion
		Size2i popup_size = get_final_transform().basis_xform(Vector2i(900 * EDSCALE, 700 * EDSCALE));
		popup_centered_clamped(popup_size, 0.8);
```

`get_screen_transform` doesn't work for embedded windows because it calculates the transform between the windows viewport and the OS-screen (even if the window is deeply nested in other viewports), so it is necessary to use a function, that returns the transform between the windows embedder (could be the OS-screen or a different Viewport) and the windows viewport. `get_final_transform()` is intended to be this function.

Could it be a long-term plan to make EDSCALE work by using per-window-scaling?

Reply:
Changed all theses to `final_transform`.

Reply:
After the last rebase, `get_final_transform()` stopped working at all (for both native and embedded windows), seems like it's no longer include windows position and taking content scale into amount.

Reply:
I will look into this.

Reply:
A) `Window.get_final_transform()` is not supposed to include the windows position in its embedder and never did, as far as I am aware.
B) I need to investigate `Window::content_scale_factor` in detail when I have more time.